R/module_teal.R

```diff
@@ -121,7 +123,7 @@ ui_teal <- function(id,
 
 
 #' @rdname module_teal
-srv_teal <- function(id, modules, teal_data_rv, filter = teal_slices()) {
+srv_teal <- function(id, modules, teal_data_rv, filter = teal_slices(), lockfile_task) {
```

Comment:
I still have some problems with adding new arguments here. If you compare the _importance_ of `lockfile_task` and for instance `modules` then it's clear that those two are not equally important. The current set of args is perfect in the sense that it's nicely shows what we pass down. If we extend these with small, side features (like this one) we might end-up with a mess that would be difficult to managed and improve.
I think it can be improved with better logic encapsulation but there is still some information that we need to pass down from `init()` to `srv_teal()`.

How about the below:
- create a new shiny module (!) that returns a path to the file as a reactive value (!)
- you will call that module inside `init()` and store what's returned in `session$userData$something`. It's a non-standard way of working and that's the weakest part but I think it's better than extending args.
- finally, inside `srv_teal()` you will read that `session$userData$something` and use it in download-handler. It's a reactive value so you will be able to do some JS magic like activate/deactivate or something similar.

Advantages over the current approach:
- it will encapsulate all this logic nicely in one function (module) compared to task invoke in `init()` and task process in `srv_teal()`. This would be a nice modular way of organising the logic and code. Not perfect though as the download button and its handler still needs to be in `srv_teal()`
- the logic of user provided file will be implemented only in one place / function that returns the path - currently you have it in both init and srv_teal
- it will be easier to control conditions (like rcmdcheck, test etc.) when everything will be kept together inside one module - right now we have it in both init and srv_teal
- maybe better to document?

Reply:
This is a fine alternative to be researched. Happy to try this out and compare with what we currently have. So the biggest difference is:  do not store `lockfile_task` in an object and do not pass it through `init -> srv_teal_with_splash -> srv_teal` but store `lockfile_task` in `session$userData` information and use it inside `srv_teal`. Alrighty, let's try and see how it goes!

Reply:
Totally. Duly noted!

Reply:
There is one caveat - a file name. Typically `write()` function returns a path and that path needs to be passed down to the `read()` function. You are assuming that this will be hard-coded and I think it's an acceptable solution. I like this more than extending the args.

Reply:
Adjusted the code so it assumes file creation is equivalent to the task finalization - 9edbe552b8ae125779cda56f1ca00c832d7f332a

Reply:
Do you still see any benefits of packing this functionality into a shiny module?

Reply:
thanks @pawelru for brainstorming and reviewing